src/app/features/tech-record/create/components/batch-create/batch-create.component.ts

```diff
@@ -0,0 +1,156 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { AbstractControl, FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { GlobalError } from '@core/components/global-error/global-error.interface';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { FormNodeWidth } from '@forms/services/dynamic-form.types';
+import { CustomValidators } from '@forms/validators/custom-validators';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { combineLatest, debounceTime, forkJoin, Subject, take, takeUntil, withLatestFrom } from 'rxjs';
+@Component({
+  selector: 'app-batch-create',
+  templateUrl: './batch-create.component.html'
+})
+export class BatchCreateComponent implements OnInit, OnDestroy {
+  readonly maxNumberOfVehicles = 39;
+  private destroy$ = new Subject<void>();
+  form: FormGroup;
+
+  constructor(
+    private fb: FormBuilder,
+    private globalErrorService: GlobalErrorService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private technicalRecordService: TechnicalRecordService
+  ) {
+    this.form = this.fb.group({
+      numberOfVehicles: [0, [Validators.max(39)]],
+      vehicles: this.fb.array([])
+    });
+  }
+
+  private startingObject = { vin: '' };
+
+  ngOnInit() {
+    this.technicalRecordService.editableVehicleTechRecord$.pipe(take(1)).subscribe(vehicle => {
+      if (!vehicle) {
+        this.back();
+      }
+    });
+
+    combineLatest([this.technicalRecordService.batchVehicles$, this.technicalRecordService.batchCount$, this.technicalRecordService.batchId$])
```

Comment:
this.technicalRecordService.batchId$ wants swapping out for this.TechnicalRecordService.generateNumbers$ othewise when the fields aren't filled when you click manage vehicles when generating Id's

Reply:
done. Thanks!